src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -204,10 +207,11 @@ def validate(self, data):
             # error based on the Widget's type
             data["discover_query_error"] = {"fields": f"Invalid fields: {err}"}
 
-        try:
-            builder.resolve_orderby(orderby)
-        except (InvalidSearchQuery) as err:
-            data["discover_query_error"] = {"orderby": f"Invalid orderby: {err}"}
+        if not is_equation(orderby.lstrip("-")):
```

Comment:
Does auto_add=True in the equation config not work? So that we don't have to skip orderby validation here?

Reply:
So I tried the other way with using `auto_add=True` and I found my test would still fail. I don't remember if we did something slightly different when we paired but for validating the widget we need to save `orderby: 'equation|blah'` and `auto_add` seems to just support equations in fields

Reply:
I made some changes to support this. We agreed that this change was not ideal because it would allow users to create widgets with invalid equations. I've added a test for that and used `auto_add` to account for the custom equation and done a similar reconstruction as the frontend.